### PR TITLE
WIP: Break finalizeKind into smaller functions 

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

## Proposed Changes

- break `finalizeKind` into smaller chuncks

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/broker/broker.go [hunk 1]
```diff
@@ -349,6 +349,10 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return controller.NewRequeueAfter(5 * time.Second)
 	}
 
+	return r.finalizeBrokerResources(ctx, broker, err, logger)
+}
+
+func (r *Reconciler) finalizeBrokerResources(ctx context.Context, broker *eventing.Broker, err error, logger *zap.Logger) reconciler.Event {
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:355]: err as input param?

File: control-plane/pkg/reconciler/broker/broker.go [hunk 2]
```diff
@@ -349,6 +349,10 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return controller.NewRequeueAfter(5 * time.Second)
 	}
 
+	return r.finalizeBrokerResources(ctx, broker, err, logger)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:352]: The name is very generic and it doesn't much to "FinalizeKind" what's special about the method?

Comment (matzew) [reply control-plane/pkg/reconciler/broker/broker.go:352]: oh, I removed the WIP :-)  this is a start... it's a bit messy func, the `finalizeKind`, hence this attempt

Comment (pierDipi) [reply control-plane/pkg/reconciler/broker/broker.go:352]: Yeah, I like it, I was challenging to find a better name 

somethink like `finalizeKafkaClusterResources(ctx, logger, broker)` ?

**Pull request closed without merging**